diff -urN linux-3.7.1/arch/arm/mach-pxa/z2.c linux-3.7.1-z2/arch/arm/mach-pxa/z2.c
--- linux-3.7.1/arch/arm/mach-pxa/z2.c	2012-12-17 14:14:54.000000000 -0500
+++ linux-3.7.1-z2/arch/arm/mach-pxa/z2.c	2013-01-09 03:31:43.353692158 -0500
@@ -41,6 +41,9 @@
 #include <linux/platform_data/mmc-pxamci.h>
 #include <linux/platform_data/keypad-pxa27x.h>
 #include <mach/pm.h>
+#include <mach/pxa27x-udc.h>
+#include <mach/udc.h>
+#include <mach/ohci.h>
 
 #include "generic.h"
 #include "devices.h"
@@ -676,6 +679,47 @@
 static inline void z2_pmic_init(void) {}
 #endif
 
+/*
+ * USB Switch
+ */
+static struct platform_device z2_usb_switch = {
+	.name              = "z2-usb-switch",
+	.id                = -1,
+};
+
+/*
+ * USB Gadget
+ */
+#if defined(CONFIG_USB_GADGET_PXA27X) \
+	|| defined(CONFIG_USB_GADGET_PXA27X_MODULE)
+static int z2_udc_is_connected(void)
+{
+	return 1;
+}
+
+static struct pxa27xx_udc_mach_info z2_udc_info __initdata = {
+	.udc_is_connected           = z2_udc_is_connected,
+	.gpio_pullup                = -1,
+};
+
+static void __init z2_udc_init(void)
+{
+	pxa_set_udc_info(&z2_udc_info);
+}
+#else
+static inline void z2_udc_init(void) {}
+#endif
+
+/*
+ * USB Host (OHCI)
+ */
+static struct pxaohci_platform_data z2_ohci_platform_data = {
+	.port_mode         = PMM_PERPORT_MODE,
+	.flags             = ENABLE_PORT2 | NO_OC_PROTECTION,
+	.power_on_delay    = 10,
+	.power_budget      = 500,
+};
+
 #ifdef CONFIG_PM
 static void z2_power_off(void)
 {
@@ -701,10 +745,12 @@
 	pxa_set_ffuart_info(NULL);
 	pxa_set_btuart_info(NULL);
 	pxa_set_stuart_info(NULL);
+	pxa_set_ohci_info(&z2_ohci_platform_data);
 
 	z2_lcd_init();
 	z2_mmc_init();
 	z2_mkp_init();
+	z2_udc_init();
 	z2_i2c_init();
 	z2_spi_init();
 	z2_nor_init();
@@ -712,6 +758,7 @@
 	z2_leds_init();
 	z2_keys_init();
 	z2_pmic_init();
+	z2_usb_switch_init();
 
 	pm_power_off = z2_power_off;
 }
