*** a/arch/arm/mach-pxa/z2.c	2011-11-11 15:26:32.115559561 -0600
--- b/arch/arm/mach-pxa/z2.c	2011-11-11 15:31:07.967261252 -0600
***************
*** 168,177 ****
  		.offset	= 0x40000,
  		.size	= 0x20000,
  	}, {
! 		.name	= "Flash",
! 		.offset	= 0x60000,
! 		.size	= 0x780000,
! 	}, {
  		.name	= "U-Boot Logo",
  		.offset = 0x7e0000,
  		.size	= MTDPART_SIZ_FULL,
--- 168,181 ----
  		.offset	= 0x40000,
  		.size	= 0x20000,
  	}, {
!                 .name   = "Kernel",
!                 .offset = 0x60000,
!                 .size   = 0x200000,
!         }, {
!                 .name   = "Filesystem",
!                 .offset = 0x260000,
!                 .size   = 0x580000,
!         }, {
  		.name	= "U-Boot Logo",
  		.offset = 0x7e0000,
  		.size	= MTDPART_SIZ_FULL,
